setuptools_rust/build.py

```diff
@@ -665,9 +531,6 @@ def _prepare_build_environment(cross_lib: Optional[str]) -> Dict[str, str]:
         }
     )
 
-    if cross_lib:
-        env.setdefault("PYO3_CROSS_LIB_DIR", cross_lib)
```

Comment:
Correction: this does not "force" `PYO3_CROSS_LIB_DIR` - `setdefault` will not overwrite any value already in the environment. So build environments should be untainted by this.

Whether `setuptools-rust` should be attempting to set this is a separate question. I think it's reasonable for it to make an educated guess if it's on balance helpful for users. As this value comes from the sysconfig, it seems reasonable to me to use that as a first guess.